pandas/io/parsers.py

```diff
@@ -1118,33 +1159,36 @@ def _make_engine(self, engine="c"):
             else:
                 raise ValueError(
                     f"Unknown engine: {engine} (valid options "
-                    'are "c", "python", or "python-fwf")'
+                    'are "c", "python", "arrow", or "python-fwf")'
                 )
             self._engine = klass(self.f, **self.options)
 
     def _failover_to_python(self):
         raise AbstractMethodError(self)
 
     def read(self, nrows=None):
-        nrows = _validate_integer("nrows", nrows)
-        ret = self._engine.read(nrows)
+        if self.engine == "pyarrow":
+            df = self._engine.read()
+        else:
+            nrows = _validate_integer("nrows", nrows)
+            ret = self._engine.read(nrows)
 
-        # May alter columns / col_dict
-        index, columns, col_dict = self._create_index(ret)
+            # May alter columns / col_dict
```

Comment:
I would move this to a read itself (may require a bit of refactoring the other engines), ideally here we just have

```
def read(self. nrows=None):
     nrows = ...
     df = self._engine.read(nrows)
     # continue on line 1193
```